app/code/Magento/MediaContentApi/Api/ExtractAssetFromContentInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentApi\Api;
+
+use Magento\MediaGalleryApi\Api\Data\AssetInterface;
+
+/**
+ * Used for extracting media asset list from a media content by the search pattern.
+ * @api
+ */
+interface ExtractAssetFromContentInterface
+{
+    /**
+     * Search for the media asset in content and extract it providing a list of media assets.
+     *
+     * @param string $content
+     * @return AssetInterface[]
+     */
+    public function execute(string $content): array;
```

Comment:
what's $content in this case?

Reply:
A "content" in the scope of the MediaContent domain is an alias to any kind of an object property which possibly contains a media asset. For example, a product object has a description which probably includes a media asset. A category object description and image also can have a media asset as a part of storage data.

Reply:
Data, selected for this property, is a developer choice made on next conclusion: if an object has a property which possibly contains a media asset, to get it from this property it needs to be sent to this Interface implementation. For example, currently, we represented a declaration of content properties in di:
https://github.com/magento/magento2/pull/27536/files/b6e79cf870273936cb70ffc05add1fa73a9e3334#diff-bcf67dbc51c19997838fdb60cee182abR9-R16

List of aliases, in that case, is a list of properties declared manually based on data structure and a possibility to have a media asset in properties.

Reply:
@buskamuza that's a parser that looks for image mentions in any content/text/string. Should we change the naming to make it better understandable or improving comments would be enough?